requirement-generator/requirement-satisfiable-course-generator.ts

```diff
@@ -17,25 +24,50 @@ function courseCanSatisfyCollegeOrMajorRequirement(
 
 const produceSatisfiableCourseIdForCollegeOrMajorRequirement = (
   collegeRequirement: CollegeOrMajorRequirement
-): readonly string[] => {
-  const set = new Set<string>();
+): EligibleCourses => {
+  const allMap: { [semester: string]: { [subject: string]: string[] } } = {};
   Object.entries(filteredAllCourses).forEach(([semester, courses]) => {
+    const semesterMap: { [subject: string]: string[] } = {};
     courses
       .filter(course => courseCanSatisfyCollegeOrMajorRequirement(course, collegeRequirement))
-      .forEach(it => set.add(`${semester}: ${it.subject} ${it.catalogNbr}`));
+      .forEach(course => {
+        let subjectSet = semesterMap[course.subject];
+        if (subjectSet == null) {
```

Comment:
would it be null or undefined?

Reply:
`bla == null` is equivalent to `bla === null || bla === undefined`